### PR TITLE
Shared storage mount implementation in server mode #5014

### PR DESCRIPTION
**Pull request by nikhil-mohite**

### CODE REVIEW

**Review by akshay-joshi — CHANGES_REQUESTED** [review]

File: web/pgadmin/misc/file_manager/__init__.py [hunk 1]
```diff
@@ -737,28 +763,51 @@ def validate_request(self, capability):
...
+            selectedDir = selectedDirList[0] if len(
+                selectedDirList) == 1 else None
+
+            if selectedDir:
+                if selectedDir[
+                    'restricted_access'] and not current_user.has_role(
+                        "Administrator"):
+                    raise PermissionError(gettext(
+                        "Access denied: This is restricted folder "
+                        "you are not allow to {0}.".format(mode)))
```
[the remark this answers is not in the document]
Comment (nikhil-mohite) [reply web/pgadmin/misc/file_manager/__init__.py]: Updated the error msg

File: web/pgadmin/misc/file_manager/__init__.py [hunk 2]
```diff
@@ -228,6 +236,7 @@ def init_filemanager():
                 "platform_type": data['platform_type'],
                 "show_volumes": data['show_volumes'],
                 "homedir": data['homedir'],
+                'ss': data['ss'],
```
[the remark this answers is not in the document]
Comment (nikhil-mohite) [reply web/pgadmin/misc/file_manager/__init__.py]: updated variable name and all its occurrences

File: web/pgadmin/browser/__init__.py [hunk 3]
```diff
@@ -599,6 +599,10 @@ def utils():
         auth_source = session['auth_source_manager'][
             'source_friendly_name']
 
+    shared_storage_list = []
+    if config.SERVER_MODE:
+        shared_storage_list = [sdir['name'] for sdir in config.SHARED_STORAGE]
```
[the remark this answers is not in the document]
Comment (nikhil-mohite) [reply web/pgadmin/browser/__init__.py]: Added extra logic to check the config provided by user is valid or not

File: web/config.py [hunk 4]
```diff
@@ -832,6 +832,17 @@
...
 ENABLE_BINARY_PATH_BROWSING = False
 
+##########################################################################
+# SHARED_STORAGE setting is used to enable the shared storage in server
+# mode. Provide the name, path and restricted_access values which should be
+# shared between users. If restricted_access is set to True the non admin
+# users will not able to upload/add, delete or rename file/folders in shared
+# storage, only admin user can do it. User need to provide the absolute path
+# of the folder and name can be anything that user can see on UI.
+# [{name: 'Shared 1', path: '/shared_folder', restricted_access: True/False}]
```
[the remark this answers is not in the document]
Comment (nikhil-mohite) [reply web/config.py]: Done

**Pull request merged**